### PR TITLE
sync: Add a new flag to enforce syncing only signed images, closes #455

### LINKED ISSUE #455: sync: Add a flag for syncing only signed images
Add a new flag in sync registry configuration to enforce syncing only signed images.

### PR DESCRIPTION
**Pull request by eusebiu-constantin-petu-dbk**
sync: When checking if a image is already synced also check for changes in upstream signatures.

Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/config-sync.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
 				"certDir": "/home/user/certs",
 				"maxRetries": 3,
 				"retryDelay": "5m",
+				"OnlySigned": true,
```
Comment (rchincha) [inline examples/config-sync.json:25]: "onlySigned" instead?

File: examples/README.md [hunk 2]
```diff
@@ -390,6 +390,7 @@ Configure each registry sync:
 				"certDir": "/home/user/certs",      # use certificates at certDir path, if not specified then use the default certs dir
 				"maxRetries": 5,                    # maxRetries in case of temporary errors (default: no retries)
 				"retryDelay": "10m",                # delay between retries, retry options are applied for both on demand and periodically sync and retryDelay is mandatory when using maxRetries.
+				"OnlySigned": true,                 # sync only signed images (either notary or cosign)
```
Comment (rchincha) [inline examples/README.md:393]: here also, "onlySigned"?

File: pkg/extensions/sync/signatures.go [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+	getCosignManifestURL.Path = path.Join(getCosignManifestURL.Path, "v2", repo, "manifests", cosignTag)
+
+	getCosignManifestURL.RawQuery = getCosignManifestURL.Query().Encode()
+
+	resp, err := client.R().Get(getCosignManifestURL.String())
+	if err != nil {
+		log.Error().Err(err).Str("url", getCosignManifestURL.String()).
+			Msgf("couldn't get cosign manifest: %s", cosignTag)
+
+		return m, "", err
```
Comment (rchincha) [inline pkg/extensions/sync/signatures.go:37]: return nil, "", err whenever ERROR

File: pkg/extensions/sync/signatures.go [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+	return m, cosignManifestDigest, nil
+}
+
+func getNotaryRefs(client *resty.Client, regURL url.URL, repo, digest string, log log.Logger) (ReferenceList, error) {
+	var referrers ReferenceList
+
+	getReferrersURL := regURL
+
+	// based on manifest digest get referrers
+	getReferrersURL.Path = path.Join(getReferrersURL.Path, "oras/artifacts/v1/", repo, "manifests", digest, "referrers")
```
Comment (rchincha) [inline pkg/extensions/sync/signatures.go:66]: Make sure this is a constant under routes.go etc "oras/artifacts/v1/"

**Comment by peusebiu** [comment]
> Too many things happening in this PR?
> 
> 1. Linter changes
> 2. constants (so currently as you have it, we will have pkg/api/constants/, maybe pkg/storage/constants/?)
>    Why not have pkg/api/constants.go itself?
> 3. The actual flag
> 
> Let's split it like above in 3 separate commits in this PR

Ok, I will split it into 3 commits
2. I can not put it into pkg/api/constants.go because it will result in circular import when importing pkg/api in sync.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**